src/Sulu/Bundle/SnippetBundle/Tests/Unit/Content/SnippetDataProviderTest.php

```diff
@@ -84,6 +91,79 @@ public function setUp(): void
         $this->referenceStore->getAll()->willReturn([]);
     }
 
+    public function testGetTypesConfiguration()
+    {
+        /** @var TokenStorageInterface|ObjectProphecy $tokenStorage */
+        $tokenStorage = $this->prophesize(TokenStorageInterface::class);
+        /** @var FormMetadataProvider|ObjectProphecy $formMetadataProvider */
+        $formMetadataProvider = $this->prophesize(FormMetadataProvider::class);
+
+        /** @var TokenInterface|ObjectProphecy $token */
+        $token = $this->prophesize(TokenInterface::class);
+        /** @var UserInterface|ObjectProphecy $user */
+        $user = $this->prophesize(UserInterface::class);
+
+        $tokenStorage->getToken()
+            ->shouldBeCalled()
+            ->willReturn($token->reveal());
+        $token->getUser()
+            ->shouldBeCalled()
+            ->willReturn($user);
+        $user->getLocale()
+            ->shouldBeCalled()
+            ->willReturn('en');
+
+        /** @var TypedFormMetadata|ObjectProphecy $typedFormMetadata */
+        $typedFormMetadata = $this->prophesize(TypedFormMetadata::class);
+
+        $formMetadataProvider->getMetadata('snippet', 'en', [])
+            ->shouldBeCalled()
+            ->willReturn($typedFormMetadata);
+
+        /** @var FormMetadata|ObjectProphecy $formMetadata1 */
+        $formMetadata1 = $this->prophesize(FormMetadata::class);
+        $formMetadata1->getName()
+            ->shouldBeCalled()
+            ->willReturn('template-1');
+        $formMetadata1->getTitle()
+            ->shouldBeCalled()
+            ->willReturn('translated-template-1');
+
+        /** @var FormMetadata|ObjectProphecy $formMetadata2 */
+        $formMetadata2 = $this->prophesize(FormMetadata::class);
+        $formMetadata2->getName()
+            ->shouldBeCalled()
+            ->willReturn('template-2');
+        $formMetadata2->getTitle()
+            ->shouldBeCalled()
+            ->willReturn('translated-template-2');
+
+        $typedFormMetadata->getForms()
+            ->shouldBeCalled()
+            ->willReturn([$formMetadata1, $formMetadata2]);
```

Comment:
I think it would be easier to simply use these classes, wouldn't it? They are quite simple value objects without a lot of logic. Would also guarantee that it works properly with the real implementation.